system/modules/task/actions/edit.php

```diff
@@ -206,12 +209,13 @@ function edit_POST($w) {
     }
     
     $task->fill($_POST['edit']);
+    $task->rate = abs($task->rate) == 0 ? NULL : $task->rate;
```

Comment:
Array_filter will remove null elements from the array where count will
include them. If the row on the form has three elements but one is null
(because of user permissions in my case) the form will display 2 elements
on a grid for 3. This removes nulls so that the form row can count
accurately when form elements are turned off/on.

I check the absolute value here because it is easier. But a I can see a few
problems this might cause. Thanks for pointing that out. Please reject this
request. I'll do some fixing.

Thanks

On Wed, Sep 14, 2016 at 10:33 AM, Adam Buckley <notifications@github.com>
wrote:

> In system/modules/task/actions/edit.php
> <https://github.com/2pisoftware/cmfive/pull/213#discussion_r78669817>:
>
> > @@ -206,12 +209,13 @@ function edit_POST($w) {
> >      }
> >
> >      $task->fill($_POST['edit']);
> > +    $task->rate = abs($task->rate) == 0 ? NULL : $task->rate;
>
> Why check against the absolute value of rate, but not set the rate
> variable to the absolute value? Do we want negative rates?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/2pisoftware/cmfive/pull/213/files/a0db7081daa98f7aea9915b52dc7151c39c34d60#r78669817>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIGsXEFTlu_anT7MKtNNqUVyXLgCsw-lks5qp0DjgaJpZM4J8Rd5>
> .
>